local-cli/runIOS/runIOS.js

```diff
@@ -107,22 +107,20 @@ function runOnSimulator(xcodeProject, args, scheme) {
 
     const simulatorFullName = formattedDeviceName(selectedSimulator);
     console.log(`Launching ${simulatorFullName}...`);
-    try {
-      child_process.spawnSync('xcrun', ['instruments', '-w', selectedSimulator.udid]);
-    } catch (e) {
-      // instruments always fail with 255 because it expects more arguments,
-      // but we want it to only launch the simulator
-    }
+    child_process.spawnSync('xcrun', ['simctl', 'boot', selectedSimulator.udid]);
     resolve(selectedSimulator.udid);
   })
-  .then((udid) => buildProject(xcodeProject, udid, scheme, args.configuration, args.packager, args.verbose))
-  .then((appName) => {
+  .then((udid) => new Promise((resolve) => {
+    buildProject(xcodeProject, udid, scheme, args.configuration, args.packager, args.verbose)
+      .then((appName) => resolve(udid, appName));
```

Comment:
this resolve call looks buggy

Reply:
I kind of agree, wasn't really happy with I either, but couldn't really think of a better way.. So I'm open to suggestions :)

Reply:
Oh now that I look at it again I think I can simply move the build project call into the first promise. Let me try

Reply:
I've moved the invocation of `buildProject` into the first promise. But now I wonder why it wasn't there in the first place..

Reply:
Another option would be to add the `udid` to the resolve call in `buildProject`, but that doesn't feel right either.